### PR TITLE
Adds support for PATCHing of RDF sources.

### PR DESCRIPTION
**Pull request by dbernstein**
**Implement PATCH for RDF**
* * *

**JIRA Ticket**: https://jira.lyrasis.org/browse/FCREPO-3078

* Other Relevant Links (Mailing list discussion, related pull requests, etc.)

# What does this Pull Request do?
Adds support for PATCH and enables the following integration tests: 

- FedoraLdpIT.testUpdateObjectGraph()
- FedoraLdpIT.testDeleteMultipleMultiValuedProperties
- FedoraLdpIT.testPatchBinary()
- FedoraLdpIT.testPatchBinaryDescriptionWithBinaryProperties()
- FedoraLdpIT.testPatchBinaryNameAndType()

# What's new?
Basic PATCH support with 5 old integration tests brought online.  One vestigial integration test was removed.

# How should this be tested?

```
#  create an RDF resource
curl -u fedoraAdmin:fedoraAdmin http://localhost:8080/rest/test1 -X PUT -H "Content-Type: text/turtle"

# modify  it by adding triples via SPARQL
curl -i -XPATCH -H"Content-type:application/sparql-update" -ufedoraAdmin:fedoraAdmin -d "INSERT DATA {<> <http://purl.org/dc/elements/1.1/title> 'MY TITLE'}" http://localhost:8080/rest/test1 -u fedoraAdmin:fedoraAdmin 

# verify that the change took
curl -u fedoraAdmin:fedoraAdmin http://localhost:8080/rest/test1

# modify it again by removing triples
curl -i -XPATCH -H"Content-type:application/sparql-update" -ufedoraAdmin:secret3 -d "DELETE {<> <http://purl.org/dc/elements/1.1/title> 'MY TITLE'} WHERE {}" http://localhost:8080/rest/test1 -u fedoraAdmin:fedoraAdmin
# verify that the removed triples are gone.
curl -u fedoraAdmin:fedoraAdmin http://localhost:8080/rest/test1
```

# Additional Notes:
This PR does not attempt to deal with all the potential corner cases around PATCH.  The object is to get PATCH working enough in order to unblock other work (WebACs) that depends on this functionality.

# Interested parties
Tag (@ mention) interested parties or, if unsure, @fcrepo4/committers

### CODE REVIEW

**Comment by pwinckles** [comment]
Forgot to note, I did try it locally and it worked as described.

File: fcrepo-http-api/src/main/java/org/fcrepo/http/api/ContentExposingResource.java [hunk 1]
```diff
@@ -934,12 +934,11 @@ protected static boolean isRdfContentType(final String contentTypeString) {
...
 
     protected void patchResourcewithSparql(final FedoraResource resource,
-            final String requestBody,
-            final RdfStream resourceTriples) {
+            final String requestBody) {
         updatePropertiesService.updateProperties(transaction().getId(),
                                                  getUserPrincipal(),
-                                                 resource.getId(),
-                                                 requestBody, resourceTriples);
+                                                 FedoraId.create(resource.getId()),
```
[the remark this answers is not in the document]
Comment (dbernstein) [reply fcrepo-http-api/src/main/java/org/fcrepo/http/api/ContentExposingResource.java:940]: No - same thing.  I'll change that.

[the remark this answers is not in the document]
Comment (dbernstein) [reply fcrepo-http-api/src/main/java/org/fcrepo/http/api/ContentExposingResource.java]: Fixed.

**Comment by awoods** [comment]
It looks like the following integration tests can also be restored:
- StateTokenIT:testLdpcvGetHasStateTokenRDFSource()
- StateTokenIT:testLdpcvHeadHasStateTokenRDFSource()
- StateTokenIT:testPatchWithStateTokenOnRDFSource()

**Comment by dbernstein** [comment]
@awoods :  I enabled the ignored tests.

**Pull request merged**